sherpa/astro/instrument.py

```diff
@@ -45,10 +49,12 @@
 from sherpa.data import Data1D
 from sherpa.astro import hc
 from sherpa.astro.data import DataARF, DataRMF, _notice_resp, DataIMG
+from sherpa.astro import io
 from sherpa.utils import sao_fcmp, sum_intervals, sao_arange
 from sherpa.astro.utils import compile_energy_grid
+from sherpa.models.regrid import EvaluationSpace1D
 
-WCS = None
+WCS: Optional[type["sherpa.astro.io.wcs.WCS"]] = None
```

Comment:
Well, `Union[A. B]` and `A | B` is only a syntactic change, they both have the same semantic meaning.

[the following is straying far from this PR]

What I'm interested in is when does it make sense to add typing checks (whether by mypy or some other tool) into the CI run (at the moment I don't think it's hugely useful, but...) and then, once we do, how best to do it [the latter is presumably not hard]. Maybe we'd add s separate workflow similar to `ci-pip-arch.yml` which we run "when we want" rather than as part of the main CI run. Has Astropy decided on an approach? 

Last year I think we dropped 3.8 in the 4.15.1 release (but obviously we can drop it from the main branch before that), but we need to decide how to handle the python 3.12 changes so we should get a plan together for all that stuff.